server/src/http/helpers/normalise-express-request.js

```diff
@@ -0,0 +1,32 @@
+function normaliseExpressRequest(request = {}) {
+  const hasRequiredProps = requestContainsExpressProperties(request);
+  
+  
+  if (request && hasRequiredProps) {
+    // Returns immutable request object
+    return Object.freeze({
+      method: request.method,
+      path: request.path,
+      headers: request.headers,
+      pathParams: request.params,
+      queryParams: request.query,
+      body: request.body
+    });
+  } else {
+    throw new Error('normaliseExpressRequest: An object containing Express request object properties is required');
+  }
+
+  function requestContainsExpressProperties(request) {
+    return (
+      'method' in request &&
+      'path' in request &&
+      'headers' in request &&
+      'params' in request &&
+      'query' in request &&
+      'body' in request
+    );
```

Comment:
What's the thought of using `'method' in request` rather than for example `request.method`, won't it both return a Boolean?

Reply:
Initially I did use the `request.method` way but my tests were producing inconsistent results. I think this is because a property could have a falsy value but still exist in the object, causing it to return the wrong boolean.

Reply:
Makes total sense, thanks for the clarification!